applications/IgaApplication/custom_elements/iga_base_element.h

```diff
@@ -0,0 +1,166 @@
+/*
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+//
+//  Main authors:   Thomas Oberbichler
+*/
+
+#if !defined(KRATOS_IGA_BASE_ELEMENT_H_INCLUDED)
+#define KRATOS_IGA_BASE_ELEMENT_H_INCLUDED
+
+// System includes
+#include "includes/define.h"
+#include "includes/element.h"
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+
+template <std::size_t TDofsPerNode>
+class IgaBaseElement
+    : public Element
+{
+public:
+    using IgaBaseElementType = IgaBaseElement<TDofsPerNode>;
+
+    using Vector3D = BoundedVector<double, 3>;
```

Comment:
This is misleading, the name should reflect that this is a BoundedVector!

Reply:
Thats the reason for the `...3D`. But I will change it to same as in `AMatrix`: `Vector3`